src/flood_wave_detector.py

```diff
@@ -71,8 +72,36 @@ def run(self) -> None:
         Executes the steps needed to find all the flood waves.
         """
         self.mkdirs()
-        self.find_vertices()
-        self.find_edges()
+
+        stations_life_intervals = JsonHelper.read(filepath=os.path.join(PROJECT_PATH,
+                                                                        'data', 'existing_stations.json'))
+
+        cut_dates = FloodWaveHandler.get_dates_in_between(start_date=self.start_date,
+                                                          end_date=self.end_date,
+                                                          intervals=stations_life_intervals,
+                                                          gauges=self.gauges)
+
+        for i in range(len(cut_dates) - 1):
+            gauges_copy = self.gauges
```

Comment:
Are you sure that this stores a COPY of `self.gauges` and not a REFERENCE to it? I would use `.copy()` or `copy.deepcopy(self.gauges)` from package `copy`.

Reply:
I noticed that this part should be outside of the loop anyway. But I did the deepcopy thing too.